tests/test_hydraulic_power.py

```diff
@@ -238,6 +240,243 @@ def test_hydraulic_power(self):
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="timeseries.csv",
+        )
+
+        # TODO: add check on values for hydraulic power.
+        results = solution.extract_results()
+
+        pipes = ["Pipe_7c53", "Pipe_f1a4", "Pipe_0e39", "Pipe_c50f"]
+        for pipe in pipes:
+            pipe_hp_in = results[f"{pipe}.GasIn.Hydraulic_power"]
+            pipe_hp_out = results[f"{pipe}.GasOut.Hydraulic_power"]
+            pipe_hp = results[f"{pipe}.Hydraulic_power"]
+
+            pipe_mass = results[f"{pipe}.GasIn.mass_flow"]
+
+            np.testing.assert_allclose(pipe_hp, pipe_hp_in - pipe_hp_out)
+            if pipe in ["Pipe_7c53", "Pipe_c50f"]:
+                # connected to demand, thus hydraulic_power should be 0
+                np.testing.assert_allclose(0, pipe_hp_out)
+
+            v_max = solution.gas_network_settings["maximum_velocity"]
+
+            v_inspect = results[f"{pipe}.GasOut.Q"] / solution.parameters(0)[f"{pipe}.area"]
+            v_points = [
+                i * v_max / solution.gas_network_settings["n_linearization_lines"]
+                for i in range(solution.gas_network_settings["n_linearization_lines"] + 1)
+            ]
+
+            # due to non linearity, every timestep on new linearized line, a doubled mass flow
+            # should result in more than doubled hydraulic power
+            v_inspect_line_ind = []
+            for k in range(len(v_inspect)):
+                for i in range(len(v_points)):
+                    if v_points[i] + 1e-6 < v_inspect[k] <= v_points[i + 1] + 1e-6:
+                        v_inspect_line_ind.append(i)
+            ind_check = 0
+            for k in range(len(v_inspect) - 1):
+                if v_inspect_line_ind[k] == v_inspect_line_ind[k + 1]:
+                    np.testing.assert_allclose(
+                        pipe_hp[k] * pipe_mass[k + 1] / pipe_mass[k], pipe_hp[k + 1]
+                    )
+                elif v_inspect_line_ind[k] < v_inspect_line_ind[k + 1]:
+                    np.testing.assert_array_less(
+                        (pipe_hp[k] + 1e-6) * pipe_mass[k + 1] / pipe_mass[k], pipe_hp[k + 1]
+                    )
+                    ind_check += 1
+                else:
+                    raise RuntimeError(
+                        "For this test to succeed the flow should increase over time"
+                    )
+            np.testing.assert_array_less(
+                0.5, ind_check, f"{pipe} is not checked between multiple lines"
+            )
+
+        # balance of hydraulic power
+        pipes_demand = ["Pipe_7c53", "Pipe_c50f"]
+        pipes_source = ["Pipe_0e39", "Pipe_f1a4"]
+        balance = np.zeros(3)
+        for p in pipes_demand:
+            balance += results[f"{p}.GasIn.Hydraulic_power"]
+        for p in pipes_source:
+            balance -= results[f"{p}.GasOut.Hydraulic_power"]
+        np.testing.assert_allclose(balance, 0.0, atol=1e-6)
```

Comment:
although it is/should be equivalent, I would like it if the variables on the gas_node port are checked as well